File: android/src/main/kotlin/com/sabiantech/sabian_native_common/events/AbstractMappableEvent.kt
package com.sabiantech.sabian_native_common.events

import com.sabiantech.sabian_native_common.structures.IMappable

abstract class AbstractMappableEvent<T : IMappable> : AbstractEvent() {
    fun raise(payload: T) {
        super.raise(payload.toMap)
    }
}

File: sabian_native_common_ios/ios/Classes/events/AbstractMappableEvent.swift
//
//  AbstractMappableEvent.swift
//  Runner
//
//  Created by bryosabian on 22/11/2023.
//

import Foundation


class AbstractMappableEvent<T : IMappable> : AbstractEvent {
    
    func raise(_ payload: T) {
        super.raise(payload.toMap)
    }
}
